packages/logger/src/Logger.ts

```diff
@@ -569,7 +570,10 @@ class Logger extends Utility implements ClassThatLogs {
 
     const consoleMethod = logLevel.toLowerCase() as keyof ClassThatLogs;
 
-    this.console[consoleMethod](JSON.stringify(log.getAttributes(), this.removeCircularDependencies()));
+    const isDevMode = this.getEnvVarsService().getDevMode();
```

Comment:
If implemented here this and the following line will be evaluated every time a log is about to be emitted.

Given that this is a configuration that has to be explicitly set by the developer and it lives in the environment, it's unlikely to change at runtime.

If this assumption is correct, what do you think about implementing this during class initialization in the [`setOptions` method](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/8b85db14e3cd4efbf6ec77472d97a63e036bdc2e/packages/logger/src/Logger.ts#L715) and store the indentation value as private class property (i.e. `private logIndentation: number`)?

This way the evaluation would be done once, during the init phase, and then whenever logging the value would simply be accessed.

What do you think?

Reply:
100% agree. It is always better to do "heavy" things once during the initialization phase than on every call. Thank you, I will make changes.